### PR TITLE
test(e2e): pin json matcher boundary values

### PR DESCRIPTION
**Pull request by nao1215**
Adds a self-hosted E2E suite covering the json/yaml path matcher at its boundaries: array indexing, a top-level array length, an empty array of length zero, the gt/gte/lt/lte comparators bounding a value, each scalar type (bool true/false, a float, a string carrying a quote), a four-level nested path, and the clean "selected no value" failure when a path matches nothing.

printf emits the exact document so the suite is deterministic; the matcher is OS-independent and runs on make e2e.

make test, make lint, and make e2e all pass (337 scenarios, 335 passed, 2 skipped).

## Summary by CodeRabbit

* **Documentation**
  * Expanded the end-to-end specification with a new section covering JSON matcher boundary values.
  * Updated scenario counts and contents to include the new coverage.

* **Tests**
  * Added end-to-end coverage for array indexing, array length checks, numeric comparisons, boolean/float/string equality, deeply nested path lookup, and no-match failure handling.
  * Included validation for expected failure output and exit behavior.

### CODE REVIEW

File: test/e2e/atago/json_matcher_edges.atago.yaml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+  - name: a string carrying a quote compares equal
+    steps:
+      - run: {shell: true, command: 'printf ''{"s":"a\"b"}'''}
+      - assert:
+          stdout:
+            json:
+              - path: "$.s"
+                equals: 'a"b'
+
```
[the remark this answers is not in the document]
Comment (nao1215) [reply test/e2e/atago/json_matcher_edges.atago.yaml:84]: Good catch. Fixed in #177: the scenario now writes the JSON with a byte-exact fixture and echoes it with cat, so the embedded quote no longer depends on how a printf implementation handles the unrecognized backslash escape. It was consistent on bash and dash (the CI shells), but the fixture form removes the latent cross-runner fragility entirely.
